### ISSUE TITLE
GoldenDict tray icon is misplaced in v3.5.6

### ISSUE THREAD

**Issue by shahab-vahedi** [polybar/polybar#2442]
Spoiler alert, I **think** this [PR](https://github.com/polybar/polybar/pull/2431) has caused some regression on my side.

## Checklist

* [X] I have read the appropriate section in the [contributing
  guidelines](https://github.com/polybar/polybar/blob/master/CONTRIBUTING.md)
* [X] I believe this issue is a problem with polybar itself and not a misconfiguration on my part.
* [X] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that
  may have already reported this problem.
* [X] I have checked the [known issues](https://github.com/polybar/polybar/wiki/Known-Issues) page for this problem.

## Describe the bug
After upgrading from `3.5.5` to `3.5.6`, the tray icon of `GoldenDict` is not placed correctly anymore AND renders a quarter of the screen useless.

**Expected behavior:**
GoldenDict tray in polybar **3.5.5**:

![tray_good_polybar_355](https://user-images.githubusercontent.com/16524902/120332546-60f30680-c2de-11eb-89e8-8ef53c413a15.png)

**Actual behavior:**
GoldenDict tray in polybar **3.5.6**:
![tray_bad_polybar_356](https://user-images.githubusercontent.com/16524902/120332636-7831f400-c2de-11eb-9e8d-eaa7c36f5324.png)

**Was it working before?**
* Did you also experience this bug in an earlier version of polybar? No
* If no, what was the last version where this worked correctly? **3.5.5**
I think it has something to do with this [change](https://github.com/polybar/polybar/pull/2431). It is worth mentioning though that other tray icons (`redshift-gtk` and `pidgin`) are fine though.

## To Reproduce
```
$ polybar -c /path/to/config/file example 2> /tmp/polybar.log &
$ goldendict
```

## Polybar Log
I have attached the log file for both 3.5.5 (good) and 3.5.6 (bad). The diff is interesting:
```
3.5.5:
<no entry>
3.5.6:
polybar|warn:  Systray selection already managed (window=0x4000009)

3.5.5:
polybar|error: Invalid _XEMBED_INFO for window 0x1e00003
3.5.6:
polybar|error: Failed to query _XEMBED_INFO, removing client... (Invalid _XEMBED_INFO for window 0x4000003)
```
[good.log](https://github.com/polybar/polybar/files/6576595/good.log)
[bad.log](https://github.com/polybar/polybar/files/6576596/bad.log)

## Environment:
* WM: BSPWM
* Distro: VoidLinux
* Output of `polybar -vvv`:

```
polybar 3.5.6

Features: +alsa +curl +i3 +mpd +network(wireless-tools) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: None
Compiler: /usr/lib/ccache/bin/g++
Compiler flags: -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2  -Wno-error=deprecated-declarations -g -Wall -Wextra -Wpedantic 
Linker flags: -Wl,-z,relro -Wl,-z,now -Wl,--as-needed 
```

**Comment by podiki** [comment]
I haven't used goldendict before, but installed it to test. Just trying quickly but I don't reproduce the icon problem. I'm on Arch, Intel integrated graphics, picom, XMonad.

```
╰─ polybar -vvv       
polybar 3.5.6

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/c++
Compiler flags: -D_FORTIFY_SOURCE=2 -march=native -O2 -pipe -fno-plt -Wall -Wextra -Wpedantic -O3 -DNDEBUG
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now 
```

**Comment by patrick96** [comment]
I can't reproduce this either. What version of goldendict are you running?

The thing responsible for this is some extra error checking that I added which should not accept icons into the bar if we can't get the correct information from them. However, in that case, the icon should disappear completely and not just exist somewhere on your screen (though this could also be a bug in goldendict).

Does this happen immediately when you start goldendict?

Could you also run polybar with trace logging (`-l trace`) and maybe remove all but one simple module like the date or text module (otherwise the output may be hard to parse).

**Comment by shahab-vahedi** [comment]
> What version of goldendict are you running?

It's a custom built package that is linked against `qt4` libraries:
```
$ xbps-query goldendict

architecture: x86_64
filename-sha256: 1704d2f9751d77360fbb41bfa651aa1c7d78b3e0d8dcf5421f02b4019ecbfe42
filename-size: 3410KB
homepage: http://goldendict.org/
install-date: 2020-10-07 15:12 CEST
install-script: 788B
installed_size: 8372KB
license: GPL-3.0-or-later
maintainer: John <me@johnnynator.dev>
metafile-sha256: accb85866697d4edba58a50114a09475cf15967bc9fe7411f451aa6d6cc7e3d8
pkgname: goldendict
pkgver: goldendict-1.5.0RC2_5
remove-script: 787B
repository: /void-packages/hostdir/binpkgs
run_depends:
	zlib>=1.2.3_1
	bzip2>=1.0.5_1
	lzo>=2.03_1
	libX11>=1.2_1
	libXtst>=1.0.3_1
	liblzma>=5.0.0_1
	tiff>=4.0.0_1
	libeb>=4.3.3_1
	libvorbis>=1.2.1rc1_1
	libhunspell1.7>=1.7.0_1
	libao>=1.1.0_1
	libavutil>=4.0_1
	libavformat>=4.0_1
	libavcodec>=4.0_1
	qt>=4.5.3_1
	qt-webkit>=2.3.4_7
	glibc>=2.29_1
	libstdc++>=4.4.0_1
	libgcc>=4.4.0_1
shlib-requires:
	libz.so.1
	libbz2.so.1
	liblzo2.so.2
	libX11.so.6
	libXtst.so.6
	liblzma.so.5
	libtiff.so.5
	libeb.so.16
	libvorbisfile.so.3
	libhunspell-1.7.so.0
	libao.so.4
	libavutil.so.56
	libavformat.so.58
	libavcodec.so.58
	libQtHelp.so.4
	libQtWebKit.so.4
	libQtSvg.so.4
	libQtXml.so.4
	libQtGui.so.4
	libQtNetwork.so.4
	libQtCore.so.4
	libpthread.so.0
	libstdc++.so.6
	libm.so.6
	libgcc_s.so.1
	libc.so.6
short_desc: A feature-rich dictionary lookup program
state: installed
```

> Does this happen immediately when you start goldendict?

Indeed, it happens as soon as I execute `goldendict`.

> Could you also run polybar with trace logging...?

Here comes the (trace) logging of the good (3.5.5) and the bad (3.5.6). I have only kept the `date` module as suggested.
[good.log](https://github.com/polybar/polybar/files/6580484/good.log)
[bad.log](https://github.com/polybar/polybar/files/6580485/bad.log)

**Comment by patrick96** [comment]
I was still not able to reproduce this. Likely because the goldendict I have access to is compiled against qt5 and not qt4.

But I have fixed a very similar issue for #2479 which I think also fixes your issue. Can you try out the changes in #2494 and see if that fixes it for you?

**Comment by shahab-vahedi** [comment]
@patrick96 thank you for still showing interest in this issue. Unfortunately, the problem still persists with the patch: 
[polybar_356_plus_patch.log](https://github.com/polybar/polybar/files/7149939/polybar_356_plus_patch.log)

![polybar_356_plus_patch](https://user-images.githubusercontent.com/16524902/132995731-f99dbae5-1d7f-4064-983d-6d7d5871e368.png)

**Comment by patrick96** [comment]
Are you sure you applied the patch correctly? The error message `Invalid _XEMBED_INFO for window ...` no longer exists in the patch.

**Comment by shahab-vahedi** [comment]
It was a snafu on my side, I installed the stock `polybar-3.5.6_1` package instead of the patched one. Your package **works** like a charm. I'll leave it up to you to close this issue now or when the patch is cut into a release.

**Comment by patrick96** [comment]
Nice :smiley:

I'll close it once I merge the PR. I will release 3.5.7 next week or so
